base/deprecated.jl

```diff
@@ -34,6 +35,20 @@ old (generic function with 1 method)
 julia> @deprecate old(x) new(x) false
 old (generic function with 1 method)
 ```
+
+Calls to `@deprecate` without explicit type-annotations will define deprecated methods
+accepting arguments of type `Any`. To restrict deprecation to a specific signature, annotate
+the arguments of `old`. For example,
+```jldoctest
+julia> new(x::Int) = x
+
+julia> new(x::Float64) = 2x
+
+julia> @deprecate old(x::Int) new(x)
+old (generic function with 1 method)
+```
```

Comment:
This will not pass doctesting. How about
````suggestion
```jldoctest; filter = r"in Main at.*"
julia> new(x::Int) = x;

julia> new(x::Float64) = 2x;

julia> @deprecate old(x::Int) new(x);

julia> methods(old)
# 1 method for generic function "old":
[1] old(x::Int64) in Main at deprecated.jl:70
```
````

Reply:
Ah, thanks for catching that!